core/mobile-menu.ts

```diff
@@ -0,0 +1,48 @@
+import { atom } from 'nanostores'
+
+import { fastCategories } from './fast.ts'
+import { router } from './router.ts'
+import { slowCategories } from './slow.ts'
+
+export const isMenuOpened = atom<boolean>(false)
+
+let targetRoutes = new Set(['fast', 'slow'])
+
+export function openMenu(): void {
+  if (router.get().route === 'slow') {
+    let slowCategoriesValue = slowCategories.get()
+    if (!slowCategoriesValue.isLoading && slowCategoriesValue.tree.length > 1) {
+      isMenuOpened.set(true)
+    }
+  } else if (router.get().route === 'fast') {
+    let fastCategoriesValue = fastCategories.get()
+    if (
+      !fastCategoriesValue.isLoading &&
+      fastCategoriesValue.categories.length > 1
+    ) {
+      isMenuOpened.set(true)
+    }
+  } else if (!targetRoutes.has(router.get().route)) {
```

Comment:
Why do you need this check if `targetRoutes` contains only `slow` and `fast` and you already checked them above? We can use just `} else {`